website/static/js/osfToggleheight.js

```diff
@@ -0,0 +1,105 @@
+
+/**
+* A simple UI component to use with divs that need to be collapsed or expanded.
+* Requires html element to have an ID set and have display block rule, (preferably a div tag).
+*
+* Usage:
+*
+*     $('#myPicker').osfToggleHeight({
+*         height: 100, // the height that will be visible when collapsed
+*         iconDown: 'fa fa-angle-down', // icon classes for the Down icon
+*         iconUp : 'fa fa-angle-up' // icon classes for the Up icon
+*     });
+*/
+
+'use strict';
+var $ = require('jquery');
+
+// Create the defaults once
+var pluginName = 'osfToggleHeight';
+var defaults = {
+    height:  75,
+    iconDown : 'fa fa-angle-down',
+    iconUp : 'fa fa-angle-up'
+};
+
+// The actual plugin constructor
+function ToggleHeight ( element, options ) {
+    var self = this;
+    self.element = element;
+    self.$el = $(element);
+    self.settings = $.extend( {}, defaults, options );
+    self._defaults = defaults;
+    self._name = pluginName;
+    self.init();
+}
+
+ToggleHeight.prototype.init =  function () {
+    var self = this;
+    self.collapsed = true;
+    self.showToggle = false;
+    self.gradientDiv = $('<div class="' + self.element.id + '-gradient toggle-height-gradient" style="display:none"></div>').appendTo(self.element);
```

Comment:
I'm not sure actually, I think I wanted a unique name for more refined ability to edit style down the line but that fits better as ID name so I changed these to ID names